File: merge_sorted_list/merge_sorted_list.go
package mergesortedlist

type ListNode struct {
	Val  int
	Next *ListNode
}

func mergeTwoLists(list1, list2 *ListNode) *ListNode {
	resultList := &ListNode{}

	current := resultList

	for list1 != nil && list2 != nil {
		if list1.Val < list2.Val {
			current.Next = list1
			list1 = list1.Next
		} else {
			current.Next = list2
			list2 = list2.Next
		}

		current = current.Next
	}

	if list1 != nil {
		current.Next = list1
	} else {
		current.Next = list2
	}

	return resultList.Next
}

File: merge_sorted_list/merge_sorted_list.dart
class ListNode {
  int val;
  ListNode? next;
  ListNode([this.val = 0, this.next]);
}

class Solution {
  ListNode? mergeTwoLists(ListNode? list1, ListNode? list2) {
    ListNode? resultNode = ListNode();
    ListNode? current = resultNode;

    while (list1 != null && list2 != null) {
      if (list1.val < list2.val) {
        current!.next = list1;
        list1 = list1.next;
      } else {
        current!.next = list2;
        list2 = list2.next;
      }
      current = current.next;
    }

    if (list1 != null) {
      current!.next = list1;
    } else {
      current!.next = list2;
    }

    return resultNode.next;
  }
}
